pkg/tsdb/influxdb/models/query.go

```diff
@@ -15,12 +16,62 @@ var (
 	regexpOperatorPattern           = regexp.MustCompile(`^\/.*\/$`)
 	regexpMeasurementPattern        = regexp.MustCompile(`^\/.*\/$`)
 	regexMatcherWithStartEndPattern = regexp.MustCompile(`^/\^(.*)\$/$`)
+	regexpRawQueryWhere             = regexp.MustCompile(`(?i)WHERE`)
 )
 
+func (query *Query) getRawQueryWithAdhocFilters() string {
```

Comment:
`renderTags` seems to be almost the same function. Can we make `renderTags` generic and just use that?

Reply:
This is the only comment preventing me from approving

Reply:
I'll refactor the signatures to reuse code from that function.

I like all the rest of your comments, but I don't want to resolve them in this PR, because that's all just existing code.